File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.ts
function countPairs(nums: number[], target: number): number {
  let result = 0;

  nums.sort((a, b) => a - b);

  for (let i = 0; i < nums.length - 1; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] < target) {
        result++;
      } else {
        break;
      }
    }
  }

  return result;
}

File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.swift
class Solution {
    func countPairs(_ nums: [Int], _ target: Int) -> Int {
        var result = 0
        for i in 0 ..< nums.count - 1 {
            for j in i + 1 ..< nums.count {
                let e1 = nums[i]
                let e2 = nums[j]
                
                if e1 + e2 < target {
                    result += 1
                }
            }
        }
        
        
        return result
    }
}
